core/test/unit/services/auth/api-key/admin_spec.js

```diff
@@ -0,0 +1,197 @@
+            should.not.exist(arg);
+            req.api_key.should.eql(this.fakeApiKey);
+            done();
+        });
+    });
+
+    it('shouldn\'t authenticate with missing bearer token', function (done) {
+        const token = '';
+        const req = {
+            headers: {
+                authorization: `Bearer ${token}`
+            }
+        };
+        const res = {};
+
+        authenticateAdminApiKey(req, res, function next(err) {
+            should.exist(err);
+            should.equal(err instanceof common.errors.UnauthorizedError, true);
+            err.code.should.eql('INVALID_AUTH_HEADER');
+            should.not.exist(req.api_key);
+            done();
+        });
+    });
+
+    it('shouldn\'t authenticate with broken bearer token', function (done) {
+        const token = 'invalid';
+        const req = {
+            headers: {
+                authorization: `Bearer ${token}`
+            }
+        };
+        const res = {};
+
+        authenticateAdminApiKey(req, res, function next(err) {
+            should.exist(err);
+            should.equal(err instanceof common.errors.BadRequestError, true);
+            err.code.should.eql('INVALID_JWT');
+            should.not.exist(req.api_key);
+            done();
+        });
+    });
+
+    it('shouldn\'t authenticate with invalid/unknown key', function (done) {
+        const token = jwt.sign({}, this.secret, {
+            algorithm: 'HS256',
+            expiresIn: '5m',
+            audience: '/test/',
+            issuer: 'unknown',
+            keyid: 'unknown'
+        });
+
+        const req = {
+            originalUrl: '/test/',
+            headers: {
+                authorization: `Bearer ${token}`
+            }
+        };
+        const res = {};
+
+        authenticateAdminApiKey(req, res, function next(err) {
+            should.exist(err);
+            should.equal(err instanceof common.errors.UnauthorizedError, true);
+            err.code.should.eql('UNKNOWN_ADMIN_API_KEY');
+            should.not.exist(req.api_key);
+            done();
+        });
+    });
+
+    it('shouldn\'t authenticate with content_key query param', function (done) {
+        const req = {
+            originalUrl: '/test/',
+            query: {
+                content_key: 'qwertyuiop'
+            }
+        };
+        const res = {};
+
+        authenticateAdminApiKey(req, res, function next(err) {
+            should.exist(err);
+            should.equal(err instanceof common.errors.BadRequestError, true);
+            err.code.should.eql('INVALID_AUTH_TYPE');
+            should.not.exist(req.api_key);
+            done();
+        });
+    });
+
+    it('shouldn\'t authenticate with JWT signed > 5min ago', function (done) {
+        const payload = {
+            iat: Math.floor(Date.now() / 1000) - 6 * 60
+        };
+        const token = jwt.sign(payload, this.secret, {
+            algorithm: 'HS256',
+            expiresIn: '10m',
```

Comment:
I think you can leave this as `5m` - although the arg is `expiresIn`, it gets converted to an `exp` claim which is just absolute expiry time, not relative to the issue time. Doesn't really matter but the different value caught me off guard a little

Reply:
This was a test for the `maxAge` being too long which requires `expiresIn` to be greater than 5m.

Reply:
:+1: sorry  about that